src/crypto/slow-hash.c

```diff
@@ -34,6 +34,36 @@
 #include <string.h>
 #include <stdio.h>
 #include <unistd.h>
+#ifdef _WIN32
+#include <windows.h>
+#define MUTEX_TYPE	HANDLE
+#define MUTEX_INIT	NULL
+#define MUTEX_LOCK(x)	my_mutex_lock(&x)
+#define MUTEX_UNLOCK(x)	(ReleaseMutex(x) == 0)
+int my_mutex_lock(volatile MUTEX_TYPE *mx)
+{
+  if (*mx == NULL) {
+    HANDLE p = CreateMutex(NULL, FALSE, NULL);
+    if (InterlockedCompareExchangePointer((PVOID*)mx, (PVOID)p, NULL) != NULL)
+      CloseHandle(p);
+  }
+  return WaitForSingleObject(*mx, INFINITE) == WAIT_FAILED;
+}
+#define THREAD_TYPE	HANDLE
+#define THREAD_RTYPE	void
+#define THREAD_CREATE(thr, func, arg)	thr = _beginthread(func, 0, arg)
+#define THREAD_JOIN(thr)			WaitForSingleObject(thr, INFINITE)
+#else
+#include <pthread.h>
+#define MUTEX_TYPE pthread_mutex_t
+#define MUTEX_INIT	PTHREAD_MUTEX_INITIALIZER
+#define MUTEX_LOCK(x)	pthread_mutex_lock(&x)
+#define MUTEX_UNLOCK(x)	pthread_mutex_unlock(&x)
+#define THREAD_TYPE pthread_t
+#define THREAD_RTYPE	void *
+#define THREAD_CREATE(thr, func, arg)	pthread_create(&thr, NULL, func, arg)
+#define THREAD_JOIN(thr)			pthread_join(thr, NULL)
+#endif
```

Comment:
That should go into a c_threads.[ch] or something.

Reply:
I suppose. I figured nothing else has needed it yet, so making it a separate include file doesn't add any utility. Can move it if you feel strongly about it.

Reply:
It just feels out of place. If it's too much busy work, I'm OK with it staying.